src/lib/evil/evil_libgen.h

```diff
@@ -0,0 +1,13 @@
+#ifndef __EVIL_LIBGEN_H__
+#define __EVIL_LIBGEN_H__
+
+#include <evil_private.h>
+
+#ifdef _MSC_VER
+# define basename evil_basename
+#endif
```

Comment:
My question here is, should we guard this `evil` functions with `_MSC_VER` or with `HAVE_<func>`?

Reply:
I think `HAVE_<func>` makes more sense, but how to generate `HAVE_<func>`?

Reply:
For now it seems ok to me doing the following:
```suggestion
#ifndef HAVE_BASENAME
#define HAVE_BASENAME
# define basename evil_basename
#endif
```
But in the case of using mingw, would meson define `HAVE_BASENAME` before us?

Reply:
you have to include it to `header_checks/meson.build` like in #312 ([here](https://github.com/expertisesolutions/efl/pull/312/files#diff-2f0220d6614e01cb370018711bbe493a))

Reply:
Thanks, I'll do it.

Reply:
for every `[foo, header]` on `function_checks` meson looks if there is any function `foo` at  `header` and if so it generated a `HAVE_FOO` with value 1.

Reply:
Ok, done. I kept the declaration outside of that guard because it is used in their test case.